### ISSUE TITLE
Python wrappers for some browser APIs

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#2430]
## Problem

In order to use `setTimeout` from JavaScript, we need to use `create_once_callable`:
Wont work:
```py
from js import setTimeout
async def test():
	await sleep(0.1)
	print("hi!")
setTimeout(test, 100)
```

Works:
```py
from js import setTimeout
async def test():
	await sleep(0.1)
	print("hi!")
setTimeout(create_once_callable(test), 100)
```

## 🚀 Feature

Python wrappers for `setTimeout`, `cancelTimeout`, `setInterval`, `cancelInterval`, `requestAnimationFrame`, `cancelAnimationFrame`
Possibly also `addEventListener` and `removeEventListener`.
These should automatically manage PyProxy lifetimes.

See:
https://github.com/pyodide/pyodide/blob/3f9ae2ed1e1aee688251af86a4dc23d3d867740c/packages/matplotlib/src/browser_backend.py#L18

**Comment by lesterfan** [comment]
I'm going to try working on this!

**Comment by hoodmane** [comment]
Closed by #2456.
